### PR TITLE
Optimize code: Fix Constructor to determine illegal logic problems

### PR DESCRIPTION
**Pull request by zhaoyuguang**
## What is the purpose of the change

Optimize code: Fix Constructor to determine illegal logic problems

## Brief changelog

Util class PojoUtils has Constructor to determine illegal logic problems。

## Verifying this change

image

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java [hunk 1]
```diff
@@ -515,7 +515,7 @@ private static Object newInstance(Class<?> cls) {
         } catch (Throwable t) {
             try {
                 Constructor<?>[] constructors = cls.getDeclaredConstructors();
-                if (constructors != null && constructors.length == 0) {
+                if (constructors == null && constructors.length == 0) {
```
Comment (CrazyHZM) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: `&&` should be replaced with `||`

Comment (lixiaojiee) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: right, I also missed out here.

Comment (zhaoyuguang) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: @CrazyHZM  nice

Comment (zhaoyuguang) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: i fix it

**Comment by zhaoyuguang** [comment]
> Actually, from the Javadoc of `java.lang.Class#getDeclaredConstructors` we can see that it is impossible for this method to return a null value.

Rogar~, I do the code more better.

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java [hunk 2]
```diff
@@ -515,7 +515,7 @@ private static Object newInstance(Class<?> cls) {
         } catch (Throwable t) {
             try {
                 Constructor<?>[] constructors = cls.getDeclaredConstructors();
-                if (constructors == null || constructors.length == 0) {
+                if (constructors.length == 0) {
```
Comment (ralf0131) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: Please add a comment here, why the null check has been removed.

Comment (zhaoyuguang) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: Rogar~

Comment (ralf0131) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: @satansk Agree! Line 524 should be removed.

Comment (zhaoyuguang) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/PojoUtils.java]: Yes, this is better.

**Pull request merged**